src/app-layer-ftp.c

```diff
@@ -414,15 +545,219 @@ static int FTPRegisterPatternsForProtocolDetection(void)
     return 0;
 }
 
+
+static StreamingBufferConfig sbcfg = STREAMING_BUFFER_CONFIG_INITIALIZER;
+
+/**
+ * \brief This function is called to retrieve a ftp request
+ * \param ftp_state the ftp state structure for the parser
+ * \param input input line of the command
+ * \param input_len length of the request
+ * \param output the resulting output
+ *
+ * \retval 1 when the command is parsed, 0 otherwise
+ */
+static int FTPDataParse(Flow *f, FtpDataState *ftpdata_state,
+        AppLayerParserState *pstate,
+        uint8_t *input, uint32_t input_len,
+        void *local_data, int direction)
+{
+    uint16_t flags = FileFlowToFlags(f, direction);
+    int ret = 0;
+    /* we depend on detection engine for file pruning */
+    flags |= FILE_USE_DETECT;
+    if (ftpdata_state->files == NULL) {
+        struct ftp_transfer_cmd *data = (struct ftp_transfer_cmd *)FlowGetStorageById(f, AppLayerExpectationGetDataId());
+        if (data == NULL) {
+            SCReturnInt(-1);
+        }
+
+        ftpdata_state->files = FileContainerAlloc();
+        if (ftpdata_state->files == NULL) {
+            SCLogError(SC_ERR_MEM_ALLOC, "Could not create file container");
```

Comment:
error msg in packet path, might be useful to use an event

Reply:
Just removed the log message for now. I did not find any similar event. Should we use a counter instead ?